src/lib.rs

```diff
@@ -460,12 +472,44 @@ impl Wallet {
     }
 }
 
-fn to_script_pubkey(address: &str) -> Result<Script, BdkError> {
-    Address::from_str(address)
+fn to_script_pubkey(address: &str) -> Result<BdkScript, BdkError> {
+    BdkAddress::from_str(address)
         .map(|x| x.script_pubkey())
         .map_err(|e| BdkError::Generic(e.to_string()))
 }
 
+/// A Bitcoin address.
+struct Address {
+    address: BdkAddress,
+}
+
+impl Address {
+    fn new(address: String) -> Result<Self, BdkError> {
+        BdkAddress::from_str(address.as_str())
+            .map(|a| Address { address: a })
+            .map_err(|e| BdkError::Generic(e.to_string()))
+    }
+
+    fn script_pubkey(&self) -> Arc<Script> {
+        Arc::new(Script {
+            script: self.address.script_pubkey(),
+        })
+    }
+}
+
+/// A Bitcoin script.
+#[derive(Clone)]
+pub struct Script {
```

Comment:
Since this is just a wrapper, we maybe could make this less verbose (and avoid the `script.script` parts below) by using a tuple struct?
```suggestion
pub struct Script(BdkScript);
```

Reply:
This is interesting, I didn't think of using tuple structs for those (I think we could potentially do the same thing for `Address(BdkAddress)`. Let me take a stab at refactoring and see what it looks like.